controllers/resources/manager.go

```diff
@@ -0,0 +1,207 @@
+package resources
+
+import (
+	"context"
+	"strings"
+
+	"github.com/go-logr/logr"
+	"github.com/pkg/errors"
+
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/utils/pointer"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	remediationv1alpha1 "github.com/medik8s/node-healthcheck-operator/api/v1alpha1"
+)
+
+const templateSuffix = "Template"
+
+type Manager interface {
+	GetTemplate(nhc *remediationv1alpha1.NodeHealthCheck) (*unstructured.Unstructured, error)
+	GenerateRemediationCRBase(gvk schema.GroupVersionKind) *unstructured.Unstructured
+	GenerateRemediationCR(node *corev1.Node, nhc *remediationv1alpha1.NodeHealthCheck, template *unstructured.Unstructured) *unstructured.Unstructured
+	CreateRemediationCR(remediationCR *unstructured.Unstructured, nhc *remediationv1alpha1.NodeHealthCheck, log logr.Logger) (bool, error)
+	DeleteRemediationCR(remediationCR *unstructured.Unstructured, nhc *remediationv1alpha1.NodeHealthCheck) (bool, error)
+	GetNodes(labelSelector metav1.LabelSelector) ([]corev1.Node, error)
+	GetOwnedInflightRemediations(nhc *remediationv1alpha1.NodeHealthCheck) (map[string]metav1.Time, error)
+	GetAllInflightRemediations(nhc *remediationv1alpha1.NodeHealthCheck) ([]unstructured.Unstructured, error)
+}
+
+type manager struct {
+	client.Client
+	ctx context.Context
+}
+
+var _ Manager = &manager{}
+
+func NewManager(c client.Client, ctx context.Context) Manager {
+	return &manager{
+		Client: c,
+		ctx:    ctx,
+	}
+}
+
+func (m *manager) GetTemplate(nhc *remediationv1alpha1.NodeHealthCheck) (*unstructured.Unstructured, error) {
+	t := nhc.Spec.RemediationTemplate.DeepCopy()
+	template := new(unstructured.Unstructured)
+	template.SetGroupVersionKind(t.GroupVersionKind())
+	template.SetName(t.Name)
+	template.SetNamespace(t.Namespace)
```

Comment:
I am not sure it is needed, but in fetchTemplate we also set the [APIVersion](https://github.com/medik8s/node-healthcheck-operator/blob/main/controllers/nodehealthcheck_controller.go#L485).

Reply:
it was redundant, it's set by SetGroupVersionKind above